### DISCUSSION TITLE
Projeto carrinho de compras  da erro ao executar e arquivo serve.json desconfigura no mockoon

### DISCUSSION THREAD

**Discussion by EmerApOli** [balta-io/forum#143] [Tira Dúvidas] [answered]
Fui rodar o projeto do curso:

1 
Criando Apps Web com Angular e Redux porem esta dando erro abaixo:

ERROR in src/app/app.component.ts:14:36 - error TS2345: Argument of type '"cart"' is not assignable to parameter of type '"products" | "total"'.
[ng] 14     this.cart$ = store.pipe(select('cart'));
[ng]                                       ~~~~~~
[ng] src/app/components/navbar/navbar.component.ts:18:36 - error TS2345: Argument of type '"cart"' is not assignable to parameter of type '"products" | "total"'.
[ng] 18     this.cart$ = store.pipe(select('cart'));
[ng]                                       ~~~~~~
[ng] src/app/components/shopping-cart/shopping-cart.component.ts:20:36 - error TS2345: Argument of type '"cart"' is not assignable to parameter of type '"products" | "total"'.
[ng] 20     this.cart$ = store.pipe(select('cart'));

2- problema.
quando tento adicionar o arquivo serve.json no mockoon, ele perde as configurações, vou em file -> new envioment e pego arqui

o caminho da sua versão esta difente, quando faço o ger não bate no end point   http://localhost:3000/v1/products
e desconfigura o arquivo

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @EmerApOli, como vai? Você está utilizando as mesmas versões do Angular CLI/Node que no curso? 
Precisam ser iguais para ter o mesmo resultado.

Depois você pode atualizar as versões, assim que for desenrolando no curso...
